fontforge/stemdb.c

```diff
@@ -4616,14 +4616,14 @@ static void MarkDStemCorner( struct glyphdata *gd,struct pointdata *pd ) {
     for ( i=0; i<pd->prevcnt && !has_stem; i++ ) {
 	stem = pd->prevstems[i];
 	if ( !stem->toobig && (
-	    ( x_dir && ( hv = IsUnitHV( &stem->unit,true ) == 1 )) ||
+	    ( x_dir && ( (hv = IsUnitHV( &stem->unit,true )) == 1 )) ||
 	    ( !x_dir && hv == 2 )))
```

Comment:
You're right, in C (on the contrary of C++), we don't know in which order conditions are evaluated, so I think it could be better to put "hv = IsUnitHV( &stem->unit,true )" just after "stem" assignation in both cases.